src/jit/lowerxarch.cpp

```diff
@@ -575,31 +575,6 @@ void Lowering::TreeNodeInfoInit(GenTree* stmt)
             op1 = tree->gtOp.gtOp1;
             op2 = tree->gtOp.gtOp2;
 
-            // See if we have an optimizable power of 2 which will be expanded 
-            // using instructions other than division.
-            // (fgMorph has already done magic number transforms)
-
-            if (op2->IsIntCnsFitsInI32())
-            {
-                bool isSigned = tree->OperGet() == GT_MOD || tree->OperGet() == GT_DIV;
-                ssize_t amount = op2->gtIntConCommon.IconValue();
-
-                if (isPow2(abs(amount)) && (isSigned || amount > 0)
```

Comment:
Would it be a good idea to assert that if divisor is const then it is not a power of 2 here, just as a safe guard to detect if there are any cases that lowering logic misses to lower those cases?